### ISSUE TITLE
Lsp config overriding user key mappings

### ISSUE THREAD

**Issue by Keitokuch** [AstroNvim/AstroNvim#675]
**Is your feature request related to a problem? Please describe.**
The key mappings defined for the LSP server in `lsp.on_attach()` (in configs/lsp/handlers.lua) overrides user key mappings in `polish()`

**Describe the solution you'd like**
The LSP client mappings can be left out in default configs, and suggested to be added by user in user.lsp.on_attach.
For example, move the mappings in `astronvim.lsp.on_attach` to `user_example.lsp.on_attach`

Thanks!

**Comment by mehalter** [comment]
ah I haven't thought about this. I just pushed an update with a new approach to enabling user configured mappings. This makes a table `mappings` and `lsp.mappings` in the user config one configures the global mappingg and the other allows the user to configure the mappings in the `on_attach`. This will let you directly configure/add/override mappings we set in the `on_attach`.

We really want there to be the default mappings for LSP out of the box sowe don't want to move that to the user config. Ideally a use should be able to install AstroNvim and add treesitter parsers and language servers and use that fully without ever needing to make a `user/init.lua` file if that makes sense :)
